File: 01/01.py
# part 1
from functools import reduce


def fuel(mass):
    return mass // 3 - 2


data = list(map(int, map(str.strip, open('input.txt', 'r').readlines())))
solution = reduce((lambda x, y: x + y), map(fuel, data))
print(solution)

File: 01/01.hs
-- part 1 and 2

import System.IO

solution = fmap (foldr1 (+) . map (fuel . read) . lines) (readFile "input.txt")
    where
        fuel mass
            | fuelMass <= 0 = 0
            | otherwise = fuelMass + fuel fuelMass
                where fuelMass = mass `div` 3 - 2